lib/cyanide/binary.ex

```diff
@@ -42,7 +42,22 @@ defmodule Cyanide.Binary do
           | pos_integer()
 
   @type t() :: %__MODULE__{
-          subtype: pos_integer(),
+          subtype: subtype(),
           data: binary()
         }
+
+  @spec subtype_int_to_atom(byte) :: subtype() | :error
```

Comment:
Usual Elixir pattern: if something can return `:error`, than the counterpart should be an `:ok` tagged tuple, so:
```
0 -> {:ok, :generic}
[...]
_ -> :error

Reply:
Also, since this function can actually return something that's not an atom, I'd probably use a more generic name (e.g. `cast_subtype`)